src/editor.jsx

```diff
@@ -346,16 +348,44 @@ var Editor = React.createClass({
             widgets = {};
             underlayPieces = [];
 
+            var searchResultIndex = 0;
+
             for (var i = 0; i < pieces.length; i++) {
                 var type = i % 2;
                 if (type === 0) {
                     // Normal text
-                    underlayPieces.push(pieces[i]);
+                    if (this.props.searchString !== "") {
```

Comment:
`if (this.props.searchString)` is a bit more concise here, and is probably what we want if searchString is null or undefined (we use editors in a couple other places, notably for the radio widget).